src/gluonts/itertools.py

```diff
@@ -289,6 +289,43 @@ def __iter__(self):
         return filter(self.fn, self.iterable)
 
 
+@dataclass
+class MixIterators:
```

Comment:
I don't like the name `MixIterators`.

Reply:
How about `WeightedIterables` or something with probabilistic?

Reply:
That name confuses me.

Reply:
Changed to `WeightedIterables`.